src/compiler/utilities.ts

```diff
@@ -6732,14 +6732,25 @@ namespace ts {
         return <T>changeAnyExtension(path, newExtension, extensionsToRemove, /*ignoreCase*/ false);
     }
 
-    export function tryParsePattern(pattern: string): Pattern | undefined {
-        // This should be verified outside of here and a proper error thrown.
-        Debug.assert(hasZeroOrOneAsteriskCharacter(pattern));

```

Comment:
Why did we remove the assert or requirement that users of this have this checked already?

Reply:
The check was duplicated between the caller and the assert (none needed it for other reasons - they were just doing it to satisfy this requirement).  I didn't think the requirement made the API conceptually clearer or the implementation simpler - on the contrary, I thought it was preferable to gather all the checks into a single function.

Reply:
Perhaps your question was "what happened to the proper error?".  It's still there - an `undefined` result now indicates that an error should be reported.